staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_vmss.go

```diff
@@ -86,8 +86,9 @@ func (ss *scaleSet) AttachDisk(isManagedDisk bool, diskName, diskURI string, nod
 	defer cancel()
 
 	// Invalidate the cache right after updating
-	key := buildVmssCacheKey(nodeResourceGroup, ss.makeVmssVMName(ssName, instanceID))
-	defer ss.vmssVMCache.Delete(key)
+	if err = ss.deleteCacheForNode(vmName); err != nil {
```

Comment:
looks like I got the root cause why it's dirty cache only in vmss, we should use defer here @feiskyer